tests/integration/test_debug.py

```diff
@@ -223,11 +223,11 @@ def test_tracer_loglevel_info_no_connection(self):
     )
     def test_tracer_loglevel_info_no_connection_py2_handler(self):
         tracer = ddtrace.Tracer()
-        tracer.log = mock.MagicMock()
         logging.basicConfig()
-        tracer.configure()
-        if PY2:
-            assert tracer.log.log.mock_calls == []
+        with mock.patch.object(logging.Logger, "log") as mock_logger:
```

Comment:
Do we have a test that does this kind of patching to confirm that the mock actually works? It's not obvious if this patching actually works.

Reply:
The test on line 216 (test_tracer_loglevel_info_no_connection) patches the logger and asserts it is called. So the patching should be working.

TestGlobalConfig has a lot over overlapping tests that assert the logger is not called. I tried to make minimal changes in this PR but we could condense the tracer_log level test cases to: 

```
@run_in_subprocess(
        env_overrides=dict(
            DD_TRACE_AGENT_URL="http://localhost:8126",
            DD_TRACE_STARTUP_LOGS="1",
        )
    )
    def test_tracer_loglevel_info_connection(self):
        tracer = ddtrace.Tracer()
        logging.basicConfig(level=logging.INFO)
        with mock.patch.object(logging.Logger, "log") as mock_logger:
            tracer.configure()
        assert mock.call(logging.INFO, re_matcher("- DATADOG TRACER CONFIGURATION - ")) in mock_logger.mock_calls

    @run_in_subprocess(
        env_overrides=dict(
            DD_TRACE_AGENT_URL="http://0.0.0.0:1234",
            DD_TRACE_STARTUP_LOGS="1",
        )
    )
    def test_tracer_loglevel_info_no_connection(self):
        tracer = ddtrace.Tracer()
        logging.basicConfig(level=logging.INFO)
        with mock.patch.object(logging.Logger, "log") as mock_logger:
            tracer.configure()
        # Python 2 logs will go to stderr directly since there's no log handler
        if PY3:
            assert mock.call(logging.INFO, re_matcher("- DATADOG TRACER CONFIGURATION - ")) in mock_logger.mock_calls
            assert mock.call(logging.WARNING, re_matcher("- DATADOG TRACER DIAGNOSTIC - ")) in mock_logger.mock_calls
        else:
            assert mock_logger.mock_calls == []

    @run_in_subprocess(
        env_overrides=dict(
            DD_TRACE_AGENT_URL="http://0.0.0.0:8126",
            DD_TRACE_STARTUP_LOGS="0",
        )
    )
    def test_tracer_log_disabled(self):
        logging.basicConfig(level=logging.INFO)
        tracer = ddtrace.Tracer()
        with mock.patch.object(logging.Logger, "log") as mock_logger:
            tracer.configure()
        assert mock_logger.mock_calls == []
```